src/Async.ts

```diff
@@ -1,12 +1,25 @@
 import { Effect } from './Effect'
+import { fx } from './Fx'
+import { schedule } from './Time'
+import { dispose } from './internal/disposable'
 
 type Run<A> = (abort: AbortSignal) => Promise<A>
 
 export class Async extends Effect('fx/Async')<Run<any>> { }
 
 export const run = <const A>(run: Run<A>) => new Async(run).returning<A>()
 
-export const sleep = (ms: number) => run(abort => new Promise<void>(resolve => {
-  const timeout = setTimeout(resolve, ms)
-  abort.addEventListener('abort', () => clearTimeout(timeout), { once: true })
-}))
+// TODO: Should this move to Time?
```

Comment:
Haha, I wrestled with the same.  I'm really not sure.

Here's a thought: if we had a different operation instead of sleep: `delay: <A>(mills: number, value: A) => Fx<Async | Schedule, A>`

It makes to me that `delay` represents a particular kind of async value, a scheduled async value, i.e. a value that will become available at a specified time.  Contrast that with something like an HTTP request.  The response isn't _scheduled_.  It will become available after some_unspecified_ amount of time (possibly never), rather than at a _specified_ time.

Does that mean there are 2 kinds of async operations, scheduled and unscheduled/divergent?  Or maybe that Scheduled can be implemented using Async?  I dunno 😂

Reply:
Here's another thought.  Maybe Schedule is the wrong effect, and it should be Sleep instead.  Then the defaultTime handler could implement Sleep using Async (i.e. it turns Sleep into Async).  I may try to implement that just to see how it goes.

Reply:
That seems pretty reasonable to me

Reply:
Ok, I gave it a shot.  Let me know what you think.

Reply:
I feel pretty good about this version.  I could imagine adding a Delay effect that delays a value instead of Sleep, but they're practically isomorphic, so I don't think it matters all that much.

Reply:
All of it looks pretty good to me too